extensions/guacamole-auth-wol/src/main/java/org/apache/guacamole/auth/wol/connection/WOLConnection.java

```diff
@@ -0,0 +1,210 @@
+import org.apache.guacamole.form.TextField;
+import org.apache.guacamole.wol.WOLException;
+import org.apache.guacamole.net.auth.Connection;
+import org.apache.guacamole.net.auth.DelegatingConnection;
+import org.apache.guacamole.wol.WakeOnLAN;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A Connection which delegates functionality to an underlying connection, but
+ * which supplies attributes specific to Wake-on-LAN functionality.
+ */
+public class WOLConnection extends DelegatingConnection {
+    
+    /**
+     * The logger for this class.
+    */
+    private static final Logger logger = LoggerFactory.getLogger(WOLConnection.class);
+
+    /**
+     * The name of the attribute for the MAC address of this host.
+     */
+    public static final String WOL_ATTRIBUTE_MAC_ADDRESS = "wol-mac-address";
+
+    /**
+     * The name of the attribute that stores the network address to use
+     * to broadcast the WOL packet.
+     */
+    public static final String WOL_ATTRIBUTE_NET_BROADCAST = "wol-net-broadcast";
+
+    /**
+     * The default broadcast address to use if none is configured.
+     */
+    public static final String WOL_DEFAULT_BROADCAST = "255.255.255.255";
+    
+    /**
+     * The form containing Wake-on-LAN attributes.
+     */
+    public static final Form WOL_ATTRIBUTE_FORM = new Form("wol-attributes",
+        Arrays.asList(
+                new TextField(WOL_ATTRIBUTE_MAC_ADDRESS),
+                new TextField(WOL_ATTRIBUTE_NET_BROADCAST)
+        )
+    );
+    
+    /**
+     * The collection of forms for Wake-on-LAN configuration.
+     */
+    public static final Collection<Form> ATTRIBUTES =
+            Collections.unmodifiableCollection(Arrays.asList(
+                    WOL_ATTRIBUTE_FORM
+        )
+    );
+
+    /**
+     * The array containing all of the WOL-specific attributes.
+     */
+    public static final List<String> WOL_ATTRIBUTES = Arrays.asList(
+            WOL_ATTRIBUTE_MAC_ADDRESS,
+            WOL_ATTRIBUTE_NET_BROADCAST
+    );
+
+    /**
+     * The attributes for this connection.
+     */
+    private final Map<String, String> attributes;
+
+    /**
+     * The undecorated connection object used as the base of this WOLConnection.
+     */
+    private final Connection undecorated;
+    
+    /**
+     * Whether or not the user has access to update the data for this connection.
+     */
+    private final Boolean canUpdate;
+
+    /**
+     * Create a WOLConnection with the specified Connection object as the
+     * base, and indicate whether the user has privileges to update this
+     * connection.
+     *
+     * @param connection
+     *     The Connection object to decorate.
+     * 
+     * @param canUpdate
+     *     Whether or not the user can update this connection.
+     */
+    public WOLConnection(Connection connection, Boolean canUpdate) {
+
+        super(connection);
+        this.undecorated = connection;
+        this.attributes = super.getAttributes();
```

Comment:
Caching the attributes of the superclass could be problematic. I would recommend instead calling `getAttributes()` where needed, saving the result of that call only locally to the function that needs it.